### DISCUSSION TITLE
‼️Upcoming 9.0.0 release will have multiple breaking changes‼️

### DISCUSSION THREAD

**Discussion by mfn** [rebing/graphql-laravel#969] [General]
👋🏼 

- the upgrade to graphql-php 15 brings upstream breaking changes
  => https://github.com/rebing/graphql-laravel/pull/953
- removing GraphQL type eager loading (i.e. removing support for non-lazy loaded types) removes some weird things we had in the code base, especially tests, and I think it's time to get rid of it and just always use a type loader
  => https://github.com/rebing/graphql-laravel/pull/966
- say goodbye to Laravel 6. With L10 around the corner, I think it's about time.
  => https://github.com/rebing/graphql-laravel/pull/967
- Remove integrated GraphiQL support in favour of https://github.com/mll-lab/laravel-graphiql
  => https://github.com/rebing/graphql-laravel/pull/986

The graphql-php upgrade exposed some very strange type resolve errors with the eager/lazy loading I could not resolve in hours of time, so I decided to solve this the other way around and get rid of the global flag.

This further motivated me to look and realize that L6 is out of security updates and most library already removed support for it; and given that L10 was around, I'm thinking this is justified.

My current plan is to make the current 8.5.0 release the latest 8.x branch and merge all the above PRs into master and release RC1.

8.x won't be maintained after the 9.0.0 release, but I'll keep a respective branch around for a while for urgent hotfixes.

I plan to release 9.0.0 more or less "now", but let's see the feedback/arguments here! Also if you have other ideas for breaking changes, now is probably a good time 😄 

cheers

**Comment by tm1000** [comment]
Was just going to ask about this in issues! Glad I looked here instead!!

**Comment by tm1000** [comment]
I noticed that webonyx now supports standard php enums, https://webonyx.github.io/graphql-php/type-definitions/enums/#construction-from-php-enum

Is there a thought for how this library will implement them or would it be best to use just the webonyx methods. What I mean by this is that right now Enums extend `Rebing\GraphQL\Support` which extends `GraphQL\Type\Definition\Type`.

**Reply by mfn** [reply]
Good question, basic answer: I've no idea (yet).

Sadly for me, PHP 7.4 is still the base line on most projects I work and all of this looks like to be PHP 8+ exclusively.

This was kind of the technical answer. The philosophical answer would be that I see graphql-laravel as a _thin and transparent_ layer between the two libraries, therefore: the less specific code it would need to make it work, the better (IMHO).

> thin and transparent

I think this is also evident by many of the things from graphql-php showing through.

**Comment by tm1000** [comment]
https://github.com/rebing/graphql-laravel/tree/mp-graphql-php-v15

**Comment by mfn** [comment]
I've just added another item to the BC / removal list:
> - Remove integrated GraphiQL support in favour of https://github.com/mll-lab/laravel-graphiql
>   => https://github.com/rebing/graphql-laravel/pull/986

The integration is definitely very convenient, but the way it's integrated it's hard to maintain.

I always favoured a "source local" solution, that's why I kept it so loing.

But https://github.com/mll-lab/laravel-graphiql , whilst defaulting to load remote resources, also allows them to download locally; which I believe is great approach.

**Comment by mfn** [comment]
I've pushed out all the announced changes as [9.0.0-rc1](https://github.com/rebing/graphql-laravel/releases/tag/9.0.0-rc1), please head over to https://github.com/rebing/graphql-laravel/discussions/998 for any feedback from here on, thank you 🙏🏼
